internal/daemon/plugin/vendorplugin.go

```diff
@@ -36,7 +36,11 @@ type GrpcPlugin struct {
 }
 
 func (g *GrpcPlugin) Start() (string, int32, error) {
-	g.ensureConnected()
+	err := g.ensureConnected()
+	if err != nil {
+		g.log.Error(err, "failed to ensure GRPC connection on grpcPlugin start")
```

Comment:
We should construct the error instead of logging a error message at each level. I think this is what we are trying to do with this repo. @bn222 Please correct me if we aren't on the same page on this.

Reply:
Correct, the error should be passed up the stack

Reply:
Ok makes sense, updated to pass an Error back up

Reply:
looks good to me, but want others to review too before merging